### PR TITLE
HPE FlexVolume Driver Helm Chart v3.1.0

### PR DESCRIPTION
**Pull request by shivamerla**
Signed-off-by: Shiva Krishna, Merla <shivakrishna.merla@hpe.com>

### CODE REVIEW

**Comment by guangbochen** [comment]
@shivamerla can u please check the CI lint failures, thanks.

**Comment by shivamerla** [comment]
@guangbochen done. thanks.

File: charts/hpe-flexvolume-driver/v3.1.0/templates/hpe-flexvolume-plugin.yaml [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+      - name: flexvolume
+        image: {{ .Values.flexVolumeDriverImage}}: {{- .Values.flexVolumeDriverTag}}
+        imagePullPolicy: "Always"
+        lifecycle:
+          preStop:
+            # create empty file to let plugin signal handler to perform cleanup of config/cert/dory files
+            exec:
+              command: [ "/bin/sh", "-c", "touch /etc/hpe-storage/remove" ]
+        {{- if eq .Values.flavor "rke"}}
+          postStart:
```
Comment (guangbochen) [inline charts/hpe-flexvolume-driver/v3.1.0/templates/hpe-flexvolume-plugin.yaml]: using default values to install will rasing the following message, is a valid HPE backend server required for testing or a fake IP would also work? 
Exec lifecycle hook ([/bin/cp -a /etc/hpe-storage/nimble.json /var/lib/kubelet/volumeplugins/hpe.com~nimble/nimble.json]) for Container "flexvolume" in Pod "hpe-flexvolume-driver-fvq8s_kube-system(df35e7c8-0131-43a1-ad2f-d2f2a0a288ab)" failed - error: command '/bin/cp -a /etc/hpe-storage/nimble.json /var/lib/kubelet/volumeplugins/hpe.com~nimble/nimble.json' exited with 1: /bin/cp: cannot create regular file '/var/lib/kubelet/volumeplugins/hpe.com~nimble/nimble.json': No such file or directory , message: "/bin/cp: cannot create regular file '/var/lib/kubelet/volumeplugins/hpe.com~nimble/nimble.json': No such file or directory\n"

Comment (shivamerla) [reply charts/hpe-flexvolume-driver/v3.1.0/templates/hpe-flexvolume-plugin.yaml]: @guangbochen valid backend ip is required as our driver will generate certificates and login to the backend. We have tested this chart from our end with RKE.

Comment (guangbochen) [reply charts/hpe-flexvolume-driver/v3.1.0/templates/hpe-flexvolume-plugin.yaml]: Thanks can u please help to squash those commits and I will LGTM to merge, thanks.

Comment (shivamerla) [reply charts/hpe-flexvolume-driver/v3.1.0/templates/hpe-flexvolume-plugin.yaml]: @guangbochen done..thanks.

**Comment by guangbochen** [comment]
LGTM,  @deniseschannon.

**Pull request merged**